### ISSUE TITLE
Exporting via stream - not expected behaviour

### ISSUE THREAD

**Issue by kristjank** [vitaly-t/pg-promise#565]
### Expected behavior
When exporting data via stream
```
const qs = new QueryStream('select * from table')
const data = await options.database.db.stream(qs, s => s.pipe(encodeStream).pipe(snapshotWriteStream))

```
the data is exported in raw format. Meaning that the global settings for https://vitaly-t.github.io/pg-promise/global.html#event:receive should be ignored. In our implementation we have conversion of properties setup, like this: 
https://github.com/ArkEcosystem/core/blob/docs/snapshost/packages/core-database-postgres/lib/connection.js#L63

According to docs, this should be ignored when using stream export.

### Actual behaviour
What actually happens is that every 100th returned record from the query gets camelized, and others not?
{record 98: block_id: 1, sender_public_key: 'zzzz'}
{record 99: block_id: 1, sender_public_key: 'zzzz'}
{record 100: **blockId**: 1, **senderPublicKey**: 'zzzz'}
{record 101: block_id: 1, sender_public_key: 'zzzz'}

### Environment

* Version of pg-promise: latest
* OS type (Linux/Windows/Mac): mac os, linux-ubuntu
* Version of Node.js: node 10

**Comment by vitaly-t** [comment]
Why would it be ignored when [the official documentation clearly explains](http://vitaly-t.github.io/pg-promise/global.html#event:receive) that it is the very feature of the library to provide the event for both regular queries and streaming?

> Global notification of any data received from the database, coming from a regular query **or from a stream**.

**Comment by kristjank** [comment]
misread stuff while testing. 

But still why not change all records, and only the 100th ones?

**Comment by vitaly-t** [comment]
The `receive` handler receives all records. Why your code doesn't change all of them - I don't know, depends on what that code is doing.

**Comment by kristjank** [comment]
tnx, closing for now

**Comment by DanielBickler** [comment]
@vitaly-t 

I just encountered this same issue.

With my receive event setup to camelize from `receive`'s documentation and piping the stream from pg-promise to `transform` from the [csv](https://csv.js.org/) library, the 100th, 200th, etc records are camelCase but the rest are snake_case.

Using `pg-promise` 8.6.5 and `pg-query-stream` 2.0.0.

The following code reproduces the error for me:
```
import {
    DATABASE_CONNECTION_LIMIT,
    DATABASE_HOST,
    DATABASE_NAME,
    DATABASE_PASSWORD,
    DATABASE_PORT,
    DATABASE_USER
} from './configuration';
const transform = require('csv').transform;
import QueryStream from "pg-query-stream";
import Stream from "stream";
import stringify from 'csv-stringify';

export const pgp = require('pg-promise')({
    receive(data: Array<Object>) {
        camelizeColumns(data);
    }
});

const cn = {
    poolSize: DATABASE_CONNECTION_LIMIT,
    host: DATABASE_HOST,
    port: DATABASE_PORT,
    database: DATABASE_NAME,
    user: DATABASE_USER,
    password: DATABASE_PASSWORD
};
export const db = pgp(cn);

function camelizeColumns(data: Array<any>) {
    const tmp = data[0];
    for (const prop in tmp) {
        const camel = pgp.utils.camelize(prop);
        if (!(camel in tmp)) {
            for (let i = 0; i < data.length; i++) {
                const d = data[i];
                d[camel] = d[prop];
                delete d[prop];
            }
        }
    }
}

export async function testStream(): Promise<any> {
    const queryStream = new QueryStream('select * from table LIMIT 100');
    const dataStream = await new Promise<Stream.Readable>((resolve) => {
        db.stream(queryStream, (stream: Stream.Readable) => {
            resolve(stream);
        });
    });
    const transformer = transform((data: any) => {
        console.log(data.start_date_time, data.startDateTime);
        return data;
    });
    const stringifier = stringify({ header: true });
    return dataStream.pipe(transformer).pipe(stringifier);
}
```

And here are the last 5 rows of the logs (the preceding 95 rows were similar to the first 4 of these):
```
[Node] 2019-07-01T18:00:00.000Z undefined
[Node] 2019-07-01T18:00:00.000Z undefined
[Node] 2019-07-01T18:00:00.000Z undefined
[Node] 2019-07-01T18:00:00.000Z undefined
[Node] undefined 2019-07-01T18:00:00.000Z
```

Could it have something to do with resolving the stream?

**Comment by vitaly-t** [comment]
I haven't looked at it for ages. It is possible something breaking occurred between this library and the stream library. This will need to be verified.

**Comment by DanielBickler** [comment]
Thanks. Please let me know if there is anything I can do to help.

I'm not very familiar with Javascript Streams but I am willing to do what I can.

This issue is holding up a feature, is it safe to say I should work around this issue rather than waiting for a fix?

**Comment by vitaly-t** [comment]
After doing some testing, it would appear that at some point data caching logic / `batchSize` stopped working for the streams. The library was caching the data, according to the `batchSize` option inside the stream library, to provide `receive` notification of the same size.

Now that it no longer works, I had no choice but to remove the caching logic completely, because the streaming library now pushes every value into the pipe, as it receives them.

As a result, specifically for streaming, event `receive` is now invoked for every single record.

Fixed and released in version [8.7.4](https://github.com/vitaly-t/pg-promise/releases/tag/8.7.4).

@kristjank , @DanielBickler Guys, thank you for finding and reporting the issue.

**Comment by DanielBickler** [comment]
@vitaly-t Thanks for investigating and the fix, I appreciate all that you do to make this library great!
